src/Resources/public/js/Web2Print/bundle.js

```diff
@@ -11,25 +11,68 @@
  *  @license    http://www.pimcore.org/license     GPLv3 and PCL
  */
 
+pimcore.registerNS('pimcore.plugin.web2print');
 
-document.addEventListener(pimcore.events.pimcoreReady, (e) => {
-    const user = pimcore.globalmanager.get("user");
-    const perspectiveCfg = pimcore.globalmanager.get("perspective");
-
-
-    if (user.isAllowed("web2print_web2print_favourite_output_channels") && perspectiveCfg.inToolbar("settings.favorite_outputdefinitions")) {
-        let menu = pimcore.globalmanager.get("layout_toolbar").settingsMenu;
-        menu.add({
-            text: t("web2print_favorite_outputdefinitions"),
-            iconCls: "bundle_outputdataconfig_nav_icon",
-            handler: function () {
-                try {
-                    pimcore.globalmanager.get("web2print.favorite_outputdefinitions").activate();
-                }
-                catch (e) {
-                    pimcore.globalmanager.add("web2print.favorite_outputdefinitions", new pimcore.bundle.web2print.favoriteOutputDefinitionsTable());
-                }
-            }
-        });
+pimcore.plugin.web2print = Class.create({
+    getClassName: function () {
+        return 'pimcore.plugin.web2print';
+    },
+
+    initialize: function () {
+        // if the new event exists, we use this
+        if (pimcore.events.preMenuBuild) {
+            document.addEventListener(pimcore.events.preMenuBuild, this.preMenuBuild.bind(this));
+        } else {
+            document.addEventListener(pimcore.events.pimcoreReady, this.pimcoreReady.bind(this));
+        }
+    },
+
+    preMenuBuild: function (e) {
+        const perspectiveCfg = pimcore.globalmanager.get('perspective');
+
+        if(!perspectiveCfg.inToolbar('settings.favorite_outputdefinitions')){
+            return;
+        }
+
+        const user = pimcore.globalmanager.get('user');
+        if (user.admin || user.isAllowed('web2print_web2print_favourite_output_channels')) {
+            let menu = e.detail.menu.settings;
+
+            menu.items.push({
+                text: t('web2print_favorite_outputdefinitions'),
+                iconCls: 'bundle_outputdataconfig_nav_icon',
+                handler: this.openFavouriteOutputChannel
+            });
+        }
+    },
+
+    pimcoreReady: function () {
+        const perspectiveCfg = pimcore.globalmanager.get('perspective');
+
+        if(!perspectiveCfg.inToolbar('settings.favorite_outputdefinitions')){
+            return;
+        }
+
+        const user = pimcore.globalmanager.get('user');
+        if (user.admin || user.isAllowed('web2print_web2print_favourite_output_channels')) {
```

Comment:
```suggestion
        if (user.isAllowed('web2print_web2print_favourite_output_channels')) {
```
Extra check for admin isn´t needed because it gets checked in `isAllowed` method anyway.

Reply:
updated the according RPs 🍭